autopayroll-CordaService/workflows/src/main/kotlin/com/autopayroll/flows/PaymentFlow.kt

```diff
@@ -0,0 +1,67 @@
+package com.autopayroll.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.autopayroll.contracts.MoneyStateContract
+import com.autopayroll.states.MoneyState
+import com.autopayroll.states.PaymentRequestState
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.utilities.ProgressTracker
+import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.flows.FlowSession
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.flows.FinalityFlow
+import net.corda.core.flows.CollectSignaturesFlow
+
+
+
+
+// *********
+// * Flows *
+// *********
+@InitiatingFlow
+@StartableByService
+class PaymentFlowInitiator : FlowLogic<SignedTransaction>() {
+    override val progressTracker = ProgressTracker()
+
+    @Suspendable
+    override fun call(): SignedTransaction{
+        // Initiator flow logic goes here
+
+        val notary = serviceHub.networkMapCache.notaryIdentities[0]
+        val wBStateList =  serviceHub.vaultService.queryBy(PaymentRequestState::class.java).states
+        val vaultState = wBStateList.get(wBStateList.size -1).state.data
```

Comment:
Should you handle the case where this is run accidentally with no `PaymentRequestState`s?

Reply:
This flow only gets triggered when there is a PaymentRequestState. This flow is not discoverable by the RPC.